### PR TITLE
Integrate Parser Refactor

### PR DESCRIPTION
**Pull request by pefontana**
# Integrate Parser changes done in cairo-rs https://github.com/lambdaclass/cairo-rs/pull/643
# Merge after https://github.com/lambdaclass/cairo-rs/pull/643

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ default = ["pyo3/num-bigint", "pyo3/auto-initialize"]
 
 [dependencies]
 pyo3 = { version = "0.16.5" }
-cairo-rs = { git = "https://github.com/lambdaclass/cairo-rs.git", rev = "c22a11db4e22bf324534c7b6f978a40478f85d50" }
+cairo-rs = { git = "https://github.com/lambdaclass/cairo-rs.git", rev = "1908502521c615b2bf2c87b0f7afbc879e861826" }
```
Comment (pefontana) [inline Cargo.toml:16]: Todo
Update commit once https://github.com/lambdaclass/cairo-rs/pull/643 is merged

File: cairo_programs/double_value_offset.cairo [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        print("POINTER: ", ids.ptr)
+        print("POINTER WITH OFFSET: ", ids.b)
+    %}
+
+    %{
+        #print(memory.__dict__)
+        #print(ap)
+        #print("AP:" [ap + (-1)])
+        #print("AP2:" [ap + (-4)])
+    %}
```
Comment (Oppen) [inline cairo_programs/double_value_offset.cairo:19]: This looks like debug data.

File: src/ids.rs [hunk 3]
```diff
@@ -259,21 +255,13 @@ pub fn get_value_from_reference(
...
-        Some(MaybeRelocatable::Int(ref num)) => Ok(PyMaybeRelocatable::Int(num.clone())),
-        None => Err(PyValueError::new_err(
+
+    if let Some(v) = value {
+        Ok(PyMaybeRelocatable::from(v))
+    } else {
+        Err(PyValueError::new_err(
             VirtualMachineError::FailedToGetIds.to_string(),
-        )),
+        ))
```
Comment (Oppen) [inline src/ids.rs:264]:
```suggestion
    value
        .map(PyMaybeRelocatable::from)
        .ok_or_else(|_| PyValueError::new_err(
            VirtualMachineError::FailedToGetIds.to_string()
        ))
```

**Review by Oppen — APPROVED** [review]

**Review by Jrigada — APPROVED** [review]
As @Oppen said, fix those minor things and we are ready to go

File: cairo_programs/double_value_offset.cairo (hunk 2 above)
Comment (pefontana) [reply cairo_programs/double_value_offset.cairo]: jeje the first hint I need it to test, the second one have escaped me

File: src/ids.rs (hunk 3 above)
Comment (pefontana) [reply src/ids.rs]: [Done](https://github.com/lambdaclass/cairo-rs-py/pull/181/commits/88e278283054cf3d1b35ed6b86441c2f59f741e8)

**Pull request merged**